logstash-core/src/main/java/org/logstash/common/io/wip/MemoryPageIOStream.java

```diff
@@ -70,6 +71,11 @@ public MemoryPageIOStream(int capacity, byte[] initialBytes) throws IOException
     }
 
     @Override
+    public void recover() throws IOException {
+        throw new NotImplementedException();
```

Comment:
did not pay much attention to this, as it is "unused" code yet ("wip" namespace) and this is a dev-time place holder just to make sure it crashes if someone starts to work with it and run tests for example - just a friendly remainder it needs to be implemented. :P